### PR TITLE
Tutorial on analysis of new assemblies

### PR DESCRIPTION
**Pull request by nekrut**
A new tutorial on simultaneous analysis of 590 *E. coli* genomes

### CODE REVIEW

**Comment by nekrut** [comment]
@bebatut @shiltemann please take a look. I'm submitting a paper referencing this tutorial so I would appreciate quick turnaround. (I will not forget about it in Portland ;)

File: topics/assembly/tutorials/ecoli_comparison/slides.html [hunk 1]
```diff
@@ -6,49 +6,3 @@
 ---
```
Comment (bgruening) [inline topics/assembly/tutorials/ecoli_comparison/slides.html]: Can you remove this empty file?

File: topics/assembly/metadata.yaml [hunk 2]
```diff
@@ -95,8 +95,38 @@ material:
...
+        title: "Unicycler Assembly"
+        type: "internal"
+        link: "/assembly/tutorials/unicycler-assembly/tutorial.html"
+    tags:
+      - prokaryote
+    questions:
+      - "I just assembled a genome. How does it compare with already sequenced genomes?"
+      - "How do I find rearranged, inserted, or deleted regions?"
+    objectives:
+      - "Find which of already sequenced genomes are most closely related to my assembly"
```
Comment (bgruening) [inline topics/assembly/metadata.yaml]: Identification of the most closely related genome to my new assembly.

File: topics/assembly/metadata.yaml [hunk 3]
```diff
@@ -95,8 +95,38 @@ material:
...
+      - "Perform sequence comparison to locate rearrangements"
+      - "Identify genes located in deletions"
+    time_estimation: "3-4h"
+    key_points:
+      - "We learned how to download large sets of completed genomes from NCBI"
+      - "We learned how to use the new Galaxy rule-based collection builder"
```
Comment (bgruening) [inline topics/assembly/metadata.yaml]: s/new//

File: topics/assembly/metadata.yaml [hunk 4]
```diff
@@ -95,8 +95,38 @@ material:
...
+      - "We learned how to use a combination of Galaxy tools to create complex browser views of genome comparisons"
```
Comment (bgruening) [inline topics/assembly/metadata.yaml]: s/browser//

File: topics/assembly/tutorials/ecoli_comparison/data-library.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+---
+libraries:
+    # This needs to be changed to reference your Zenodo data
+    - name: "Small test files"
+      files:
```
Comment (bgruening) [inline topics/assembly/tutorials/ecoli_comparison/data-library.yaml]: can you list here all files from zenodo, preferably in the new format: https://github.com/galaxyproject/training-material/pull/759/files

File: topics/assembly/tutorials/ecoli_comparison/data-manager.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# configuration for fetch and index genomes
+---
```
Comment (bgruening) [inline topics/assembly/tutorials/ecoli_comparison/data-manager.yaml]: can you remove this file

File: topics/assembly/tutorials/ecoli_comparison/slides.html [hunk 7]
```diff
@@ -0,0 +1,8 @@
+---
```
Comment (bgruening) [inline topics/assembly/tutorials/ecoli_comparison/slides.html]: remove this one

File: topics/assembly/tutorials/ecoli_comparison/tools.yaml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+---
+api_key: admin
+galaxy_instance: http://localhost:8080
+install_resolver_dependencies: true
+install_tool_dependencies: false
+tools:
```
Comment (bgruening) [inline topics/assembly/tutorials/ecoli_comparison/tools.yaml]: This file needs to be filed. The tools are listed in the workflows with revision number.

File: topics/assembly/tutorials/ecoli_comparison/workflows/empty_workflow.ga [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
+    "a_galaxy_workflow": "true",
```
Comment (bgruening) [inline topics/assembly/tutorials/ecoli_comparison/workflows/empty_workflow.ga]: Can you add the workflow here?

File: topics/assembly/tutorials/ecoli_comparison/tutorial.md [hunk 10]
```diff
@@ -0,0 +1,965 @@
...
+
+[NCBI](https://www.ncbi.nlm.nih.gov/) is the resource that would store all complete *E. coli* genomes. Specifically, they can be found [here](https://www.ncbi.nlm.nih.gov/genome/genomes/167). As we will see, this list contains over 500 genomes and so uploading them by hand will likely result in carpal tunnel syndrome, which we want to prevent. Galaxy has several features that are specifically designed for uploading and managing large sets of similar types of data. The following two **Hands-on** sections show how they can be used to import all completed *E. coli* genomes into Galaxy.
+
+> ### {% icon hands_on %} Hands-on: Preparing a list of all complete *E. coli* genomes
+>
+>Open [the NCBI list of of *E. coli* genomes](https://www.ncbi.nlm.nih.gov/genome/genomes/167) in a new window and position two browser windows (one the tutorial and the one you just opened) side by side. Then follow the steps in the following video.
```
Comment (bgruening) [inline topics/assembly/tutorials/ecoli_comparison/tutorial.md]: We should have a tool for this in Galaxy or a data_source for Genbank.

File: topics/assembly/tutorials/ecoli_comparison/data-library.yaml (hunk 5 above)
Comment (nekrut) [reply topics/assembly/tutorials/ecoli_comparison/data-library.yaml]: I don't understand this. Is the top level thing (`destination:`) specific for the whole GTN site or just for a given tutorial?

File: topics/assembly/tutorials/ecoli_comparison/tutorial.md (hunk 10 above)
Comment (nekrut) [reply topics/assembly/tutorials/ecoli_comparison/tutorial.md]: not sure we really need this. another thing to maintain. spreadshteet works just fine

File: topics/assembly/tutorials/ecoli_comparison/workflows/empty_workflow.ga (hunk 9 above)
Comment (nekrut) [reply topics/assembly/tutorials/ecoli_comparison/workflows/empty_workflow.ga]: there is no workflow here

File: topics/assembly/tutorials/ecoli_comparison/tools.yaml (hunk 8 above)
Comment (nekrut) [reply topics/assembly/tutorials/ecoli_comparison/tools.yaml]: so I really need and catalog all tools? Man, no one will write tutorials if this is a requirement.

Comment (nekrut) [reply topics/assembly/tutorials/ecoli_comparison/tools.yaml]: Sorry for my public venting ...

Comment (bebatut) [reply topics/assembly/tutorials/ecoli_comparison/tools.yaml]: If you have a workflow, you can generate the `tools.yaml` file: see http://galaxyproject.github.io/training-material/topics/contributing/tutorials/create-new-tutorial-technical/tutorial.html#filling-the-toolsyaml

File: topics/assembly/tutorials/ecoli_comparison/workflows/empty_workflow.ga (hunk 9 above)
Comment (bgruening) [reply topics/assembly/tutorials/ecoli_comparison/workflows/empty_workflow.ga]: You could export your history and use this workflow. The workflow you could use to create the tools.yaml file. The overall idea behind workflows is that we use them for automatic testing of tutorials but also give trainees a good overview of what was/will be done.

File: topics/assembly/tutorials/ecoli_comparison/tutorial.md (hunk 10 above)
Comment (bgruening) [reply topics/assembly/tutorials/ecoli_comparison/tutorial.md]: Not for this tutorial. But imho it would be nice to have more powerful data_source tools.

**Comment by shiltemann** [comment]
@nekrut thanks a lot :) ..I just reformatted the hands-on boxes a bit, if this looks good to you I will merge it today :)

**Pull request merged**